x-pack/test/functional/apps/canvas/feature_controls/canvas_spaces.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from 'expect.js';
+import { SpacesService } from 'x-pack/test/common/services';
+import { KibanaFunctionalTestDefaultProviders } from '../../../../types/providers';
+
+// tslint:disable:no-default-export
+export default function({ getPageObjects, getService }: KibanaFunctionalTestDefaultProviders) {
+  const esArchiver = getService('esArchiver');
+  const spacesService: SpacesService = getService('spaces');
+  const kibanaServer = getService('kibanaServer');
+  const PageObjects = getPageObjects(['common', 'canvas', 'security', 'spaceSelector']);
+  const find = getService('find');
+  const appsMenu = getService('appsMenu');
+
+  describe('spaces feature controls', () => {
+    before(async () => {
+      await esArchiver.loadIfNeeded('logstash_functional');
+    });
+
+    describe('space with no features disabled', () => {
+      before(async () => {
+        // we need to load the following in every situation as deleting
+        // a space deletes all of the associated saved objects
+        await esArchiver.load('canvas/default');
+        await kibanaServer.uiSettings.replace({
```

Comment:
I found out when working on https://github.com/elastic/kibana/pull/29835 that `kibanaServer.uiSettings` doesn't work with anything outside of the default space, so we're relying on the esarchives setting this stuff, so this can be removed here.